sdks/python/apache_beam/transforms/core.py

```diff
@@ -1030,7 +1046,9 @@ def to_runner_api_parameter(self, context):
     picked_pardo_fn_data = pickler.dumps(self._pardo_fn_data())
     state_specs, timer_specs = userstate.get_dofn_specs(self.fn)
     from apache_beam.runners.common import DoFnSignature
-    is_splittable = DoFnSignature(self.fn).is_splittable_dofn()
+    do_fn_signature = DoFnSignature(self.fn)
+    is_splittable = do_fn_signature.is_splittable_dofn()
+    is_requests_finalization = do_fn_signature.is_request_finalization()
```

Comment:
We can't know that here, only once finalization callbacks have been registered.

Reply:
Yes you are right. Have changed to verify whether there are callbacks registered after finishing bundle processing.